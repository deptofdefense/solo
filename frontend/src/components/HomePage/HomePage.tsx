import React, { useCallback } from "react";
import { useHistory } from "react-router-dom";
import useAuthContext from "context/AuthContext";
import classes from "./HomePage.module.css";

const HomePage: React.FC = () => {
  const history = useHistory();
  const { username, apiLogout } = useAuthContext();

  const logoutUser = useCallback(async () => {
    await apiLogout();
    history.push("/postlogout");
  }, [apiLogout, history]);

  return (
    <div className={classes.root}>
      <h2>Protected Route</h2>
      <div>username: {username}</div>
      <button onClick={logoutUser} className="usa-button">
        Logout
      </button>
    </div>
  );
};

export default HomePage;
