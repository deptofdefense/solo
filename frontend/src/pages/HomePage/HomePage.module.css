.root {
  margin-top: 10%;
  width: 100%;
  align-items: center;
  display: flex;
  flex-direction: column;
}
