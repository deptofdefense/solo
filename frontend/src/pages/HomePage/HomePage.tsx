import React, { useCallback, useState } from "react";
import { useHistory } from "react-router-dom";
import useAuthContext from "context/AuthContext";
import { Button } from "solo-uswds";
import classes from "./HomePage.module.css";

const HomePage: React.FC = () => {
  const history = useHistory();
  const [taskId, setTaskId] = useState<null | string>(null);
  const { username, apiLogout, apiCall } = useAuthContext();

  const logoutUser = useCallback(async () => {
    await apiLogout();
    history.push("/postlogout");
  }, [apiLogout, history]);

  const createWorkerDebugMsg = useCallback(async () => {
    setTaskId(null);
    try {
      const { task_id: taskId } = await apiCall<{ task_id: string }>(
        "/workerlog/",
        {
          method: "POST",
          body: JSON.stringify({ msg: `worker debug message` })
        }
      );
      setTaskId(taskId);
    } catch (e) {
      setTaskId(e.toString());
    }
  }, [apiCall, setTaskId]);

  return (
    <div className={classes.root}>
      <h2>Protected Route</h2>
      <div>username: {username}</div>
      {taskId && <div>last task id: {taskId}</div>}
      <div>
        <Button onClick={logoutUser}>Logout</Button>
        <Button onClick={createWorkerDebugMsg}>
          Create worker debug message
        </Button>
      </div>
    </div>
  );
};

export default HomePage;
