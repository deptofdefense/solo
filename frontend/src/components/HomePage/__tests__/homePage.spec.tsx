import React from "react";
import HomePage from "../HomePage";
import { render, fireEvent } from "test-utils";

describe("home page protected route component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("includes username in the body", () => {
    const { queryByText } = render(<HomePage />, {
      authContext: {
        username: "testuser"
      }
    });
    expect(queryByText(/testuser/)).toBeInTheDocument();
  });

  it("calls logout on button click", () => {
    const fakeLogout = jest.fn();
    const { getByText } = render(<HomePage />, {
      authContext: {
        apiLogout: fakeLogout
      }
    });
    const logoutBtn = getByText("Logout");
    fireEvent.click(logoutBtn);
    expect(fakeLogout).toHaveBeenCalled();
  });
});
