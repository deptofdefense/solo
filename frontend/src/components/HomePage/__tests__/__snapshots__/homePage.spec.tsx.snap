// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page protected route component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <h2>
      Protected Route
    </h2>
    <div>
      username: 
    </div>
    <button
      class="usa-button"
    >
      Logout
    </button>
  </div>
</DocumentFragment>
`;
